core/src/main/java/com/orange/ouds/core/component/OudsBadge.kt

```diff
@@ -305,93 +358,196 @@ const val OudsBadgeMaxCount = 99
+ */
+sealed class OudsBadgeWithIconStatus(val icon: OudsBadgeIcon) {
 
-    /** Indicates success, completion, or approval. */
-    Positive,
+    /**
+     * Used for general labels without specific emphasis.
+     * Its [icon] is an [OudsBadgeIcon.Custom].
+     */
+    class Neutral internal constructor(icon: OudsBadgeIcon) : OudsBadgeWithIconStatus(icon) {
+        /**
+         * Creates an instance of [OudsBadgeStatus.Neutral] with a custom icon.
+         */
+        constructor(icon: OudsBadgeIcon.Custom) : this(icon as OudsBadgeIcon)
+    }
 
-    /** Provides informational context without urgency. */
-    Info,
+    /**
+     * Employed to highlight discovery or exploration-related content.
+     * Its [icon] is an [OudsBadgeIcon.Custom].
+     */
+    class Accent internal constructor(icon: OudsBadgeIcon) : OudsBadgeWithIconStatus(icon) {
+        /**
+         * Creates an instance of [OudsBadgeWithIconStatus.Accent] with a custom icon.
+         */
+        constructor(icon: OudsBadgeIcon.Custom) : this(icon as OudsBadgeIcon)
+    }
 
-    /** Negatives the user to potential risks or cautionary messages. */
-    Warning,
+    /**
+     * Indicates success, completion, or approval.
+     * Its [icon] is an [OudsBadgeIcon.Default].
+     */
+    class Positive internal constructor(icon: OudsBadgeIcon) : OudsBadgeWithIconStatus(icon), BadgeFunctionalStatus {
+
+        /**
+         * Creates an instance of [OudsBadgeWithIconStatus.Positive] with its default dedicated icon.
+         */
+        constructor(icon: OudsBadgeIcon.Default = OudsBadgeIcon.Default) : this(icon as OudsBadgeIcon)
+
+        @Composable
+        override fun getDefaultIconPainter() = painterResource(OudsTheme.drawableResources.success)
+    }
+
+    /**
+     * Provides informational context without urgency.
+     * Its [icon] is an [OudsBadgeIcon.Default].
+     */
+    class Info internal constructor(icon: OudsBadgeIcon) : OudsBadgeWithIconStatus(icon), BadgeFunctionalStatus {
+        /**
+         * Creates an instance of [OudsBadgeWithIconStatus.Info] with its default dedicated icon.
+         */
+        constructor(icon: OudsBadgeIcon.Default = OudsBadgeIcon.Default) : this(icon as OudsBadgeIcon)
+
+        @Composable
+        override fun getDefaultIconPainter() = painterResource(OudsTheme.drawableResources.information)
+    }
+
+    /**
+     * Negatives the user to potential risks or cautionary messages.
+     * Its [icon] is an [OudsBadgeIcon.Default].
+     */
+    class Warning internal constructor(icon: OudsBadgeIcon) : OudsBadgeWithIconStatus(icon), BadgeFunctionalStatus {
+        /**
+         * Creates an instance of [OudsBadgeWithIconStatus.Warning] with its default dedicated icon.
+         */
+        constructor(icon: OudsBadgeIcon.Default = OudsBadgeIcon.Default) : this(icon as OudsBadgeIcon)
+
+        @Composable
+        override fun getDefaultIconPainter() = painterResource(OudsTheme.drawableResources.warningExternalShape)
```

Comment:
In Figma the icon used is ".brand-theme/Component/alert/warning", which is the same as the one used for the tag, so I suppose it should use both internal and external shapes. However this is not visible in the Figma UI rendering. Do you know why?

Reply:
I think the 2 color icon is only for muted tag. Here we only display the external shape.